clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -346,6 +346,12 @@ protected void onJoinComplete(int generation,
 
         Set<TopicPartition> ownedPartitions = new HashSet<>(subscriptions.assignedPartitions());
 
+        // should at least encode the short version
+        if (assignmentBuffer.remaining() < 2)
+            throw new IllegalStateException("Coordinator returned assignment is empty, this is not expected; " +
```

Comment:
Just a nit, but there could be one byte in the message. Maybe we can phrase this message something like "there is insufficient bytes available to read version field (actual size: %d)..."

Reply:
ack.